### PR TITLE
chore: refactor external database to e2e test

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

This pull request changes the following:

* Convert external data base test from task workflow to an E2E test
* update smoke test so it can be used for different namespace
* added `process.env.SOLO_CACHE_DIR` so smoke test can pick up local config
 from `test/data/tmp` instead of the default `~/.solo`
 * add functions installPostgres, deployWithExternalDatabase and runSql to MirrorNodeTest to follow the same steps as the original task test workflow

### Related Issues

* Relates to #2080

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: test/e2e/commands/tests/relay-test.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    expect(relayPods).to.have.lengthOf(1);
+  }
+
+  public static deploy(options: BaseTestOptions): void {
+    const {testName, deployment, namespace, contexts, clusterReferenceNameArray, testLogger} = options;
+    const {soloRelayDeployArgv, verifyRelayDeployWasSuccessful} = RelayTest;
+
+    it(`${testName}: JSON-RPC relay deploy`, async (): Promise<void> => {
+      // switch back to the target cluster context
+      await new ShellRunner().run(`kubectl config use-context "${contexts[1]}"`);
```
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply test/e2e/commands/tests/relay-test.ts:54]: Removed

File: test/e2e/commands/tests/mirror-node-test.ts [hunk 2]
```diff
@@ -220,4 +225,169 @@ export class MirrorNodeTest extends BaseCommandTest {
...
+        .list(namespace, [
+          'app.kubernetes.io/instance=mirror',
+          'app.kubernetes.io/name=grpc',
+          'app.kubernetes.io/component=grpc',
+        ]);
+      const mirrorNodePod: Pod = mirrorNodePods[0];
+      await k8.pods().readByReference(mirrorNodePod.podReference).portForward(5600, 5600);
+    });
+
+    it('Enable port-forward for mirror ingress controller', async (): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply test/e2e/commands/tests/mirror-node-test.ts:299]: updated

File: test/e2e/commands/tests/mirror-node-test.ts [hunk 3]
```diff
@@ -220,4 +225,169 @@ export class MirrorNodeTest extends BaseCommandTest {
...
+  static executeBackgroundCommand(command: string, label: string, testLogger: SoloLogger): void {
+    console.log(`${label} background command:`);
+    console.log(command);
+
+    try {
+      // Remove any trailing & as we'll handle the background process ourselves
+      const cleanCommand = command.replace(/\s*&\s*$/, '');
+
+      // For background commands, we use spawn instead of execSync
+      const process = spawn(cleanCommand, {
```
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply test/e2e/commands/tests/mirror-node-test.ts:325]: removed

**Review by jeromy-cannon — APPROVED** [review]
I'm changing this to 'relates to' instead of 'closes' the GHI, as the solo smoke test shell script logic was not rewritten into the the mocha tests, but instead was just invoked as a shell.
